ComponentKitTests/Scope/CKComponentScopeTests.mm

```diff
@@ -219,4 +219,28 @@ - (void)testComponentScopeHandleGlobalIdentifierIsTheSameBetweenSiblingsWithComp
   }
 }
 
+#pragma mark - Component scope override
+
+- (void)testComponentScopeCanBeOverridden
+{
+  CKComponentScopeRoot *root = [CKComponentScopeRoot rootWithListener:nil];
+  CKThreadLocalComponentScope threadScope(root, {});
+  CKThreadLocalComponentScope *threadScopePtr = &threadScope;
+
+  dispatch_semaphore_t sema = dispatch_semaphore_create(0);
+  dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
I'm a bit concerned with this causing flakiness. I'd try following what the tests above are doing, just nesting scopes manually.